blog/posts/20251202-savetheclocktower-pulsar-on-electron-30.md

```diff
@@ -0,0 +1,82 @@
+
+  The bad news is that the `hydrogen` package is no longer maintained by its original authors; the good news is that @asiloisad, our prolific community package contributor, has built a fork called [`hydrogen-next`](https://packages.pulsar-edit.dev/packages/hydrogen-next) that ought to be a drop-in replacement.
+
+* [`x-terminal-reloaded`](https://packages.pulsar-edit.dev/packages/x-terminal-reloaded) is the only actively maintained terminal package (for now!) and is relied on by many to provide a built-in terminal. It once relied on a very old version of the `node-pty` package that was incompatible with newer Electron.
+
+  The good news is that we anticipated this a while back and worked with @Spiker985 (maintainer of `x-terminal-reloaded`) to update to a newer `node-pty` version; macOS and Linux users should experience no issues with this package as long as it’s up to date.
+
+  The bad news is that Windows users will likely run into [another issue](https://github.com/Spiker985/x-terminal-reloaded/issues/173). We’ve got [a potential fix for this issue](https://github.com/Spiker985/x-terminal-reloaded/pull/175) in the works, and if you know your way around the Node ecosystem and are a Windows user, we could use your help testing it!
+
+  The better news is that Pulsar users will soon have a much simpler option for a cross-platform terminal package, but you’ll have to wait just a bit longer for that announcement.
+
+## Other issues
+
+I’ve been running some version of Pulsar on Electron 30 for more than a year, and at least a dozen other users have been beta-testing it for months. That said, if you run into problems that aren’t covered above, please [report them in the megathread](https://github.com/pulsar-edit/pulsar/issues/1383) and we’ll get to the bottom of it.
+
+Remember that you can always revert to earlier versions of Pulsar via the [GitHub releases](https://github.com/pulsar-edit/pulsar/releases) page.
+
+## What’s next?
+
+Finally delivering this upgrade is a big load off our backs, but the job’s not done. We picked Electron 30 last year just to have a target to fixate on, but Electron’s already up to version 39! So you can expect us to gradually play catch-up over the next few releases.
+
+Luckily, future Electron upgrades should be nowhere near as disruptive as this one. The worst thing about it is that catching up to Electron 39 will drop official support for some older operating systems:
+
+* Electron 33 drops support for macOS 10.15 (Catalina).
+* Electron 38 drops support for macOS 11 (Big Sur).
+
+We regret this, but Electron apps are beholden to Chromium’s support matrix. If Chromium drops support for an operating system, it’s _dropped_. If this will affect you, you’ll at least be able to use older versions of Pulsar indefinitely. If it affects enough people, we might explore a custom release channel of Pulsar that tries to deliver new features while maintaining a broadly compatible Electron version.
```

Comment:
```suggestion
We regret this, but Electron apps are beholden to Chromium’s support matrix. If Chromium drops support for an operating system, it’s _dropped_. If this will affect you, you’ll at least be able to use older versions of Pulsar indefinitely. If it affects enough people, we might explore a custom release channel of Pulsar that tries to deliver new features while maintaining a broadly compatible Electron version. (No promises!)
```

Suggestion to add _"(No promises!)"_ after _"If it affects enough people, we might explore a custom release channel of Pulsar that tries to deliver new features while maintaining a broadly compatible Electron version."_

(Better to under-promise and/or over-deliver than to even seem to imply over-promising and/or under-delivering! Maybe! (Debatable!))

Reply:
I figured “might explore” was weasely enough, no?

Reply:
So if you're inclined against the addition we can simply mark this as resolved, no worries either way.